landscape/lib/tests/test_network.py

```diff
@@ -148,6 +148,25 @@ def test_ipv6_only_in_extended_mode(
                 'duplex': True,
                 'ip_addresses': {AF_INET6: [{'addr': '2001::1'}]}}])
 
+    @patch("landscape.lib.network.get_default_interfaces")
+    @patch("landscape.lib.network.get_network_interface_speed")
+    @patch("landscape.lib.network.get_flags")
+    @patch("landscape.lib.network.netifaces.ifaddresses")
+    @patch("landscape.lib.network.netifaces.interfaces")
+    def test_default_only_interface(
+            self, mock_interfaces, mock_ifaddresses, mock_get_flags,
+            mock_get_network_interface_speed, mock_get_default_interfaces):
+        default_iface = "test_iface"
+        mock_get_default_interfaces.return_value = [default_iface]
+        mock_get_network_interface_speed.return_value = (100, True)
+        mock_get_flags.return_value = 4163
+        mock_interfaces.return_value = [default_iface]
+        mock_ifaddresses.return_value = {AF_INET: [{"addr": "192.168.0.50"}]}
+        device_info = get_active_device_info(extended=False, default_only=True)
+        interfaces = [i["interface"] for i in device_info]
+        self.assertIn(default_iface, interfaces)
+        self.assertEqual(len(interfaces), 1)
```

Comment:
nitpick: since you properly isolated this test, the content `interfaces` should be fixed and could be a single assertEqual() on a set.